### PR TITLE
section 3: registration and segmentation with SITK

### PR DESCRIPTION
**Pull request by gcroci2**
This PR is finally ready to be reviewed :)

Few notes:
- The images used in the exercises (A1_grayT1.nrrd, A1_grayT2.nrrd, digital_xray.dcm, training_001_ct.mha, training_001_mr_T1.mha) are not on this branch, but I have them locally. If you need them to run the code, just let me know. We will publish them when we solve issue #10 
- I mentioned in the "Acknowledgements" section where I took the materials (and the images), but in those links I didn't find any information about the images they use (that I also used in this episode). I guess that they just fall under the same license as the notebooks and tutorials (creative commons and apache), but better to double check.

### CODE REVIEW

**Comment by drcandacemakedamoore** [comment]
For an outline this looks good, but did you put this PR up as a draft, or is it really to be approved by me before moving forward?

**Comment by gcroci2** [comment]
> For an outline this looks good, but did you put this PR up as a draft, or is it really to be approved by me before moving forward?

I will ask for your review when the PR is ready :) getting there!

**Comment by drcandacemakedamoore** [comment]
We should add a bell-box about different approaches to registration to highlight that it can be done with dipy (which is mentioned in the MRI episode) in very specific ways on diffusion imaging e.g. SymmetricDiffeomorphicRegistration functionality, or even bare(r) python. We must warn people to pick one system (for diffusion imaging go with dipy and for many other things go with SITK) so they don't create confusion.

**Comment by gcroci2** [comment]
> We should add a bell-box about different approaches to registration to highlight that it can be done with dipy (which is mentioned in the MRI episode) in very specific ways on diffusion imaging e.g. SymmetricDiffeomorphicRegistration functionality, or even bare(r) python. We must warn people to pick one system (for diffusion imaging go with dipy and for many other things go with SITK) so they don't create confusion. Also there is the same kind of issue with using SITK GetMetaDataKeys versus just using PyDICOM. Maybe we can have two bell boxes, or we can have one saying there is more than one way to skin a cat.

Makes sense! Maybe you can directly add a commit on this branch with it (or a scratch of it to guide me)? I think you know better what you mean and it seems to me that your vision on the table is already pretty clear :) @drcandacemakedamoore

File: episodes/simpleitk.md [hunk 1]
```diff
@@ -0,0 +1,1198 @@
...
+    image_z=(0,img_T1.GetSize()[2]-1),
+    npa = fixed(sitk.GetArrayViewFromImage(seg_img_clean)),
+    title=fixed("Confidence connected after morphological closing"))
+```
+
+![](episodes/fig/reg_grow_cleaned.png){alt='Confidence connected after morphological closing.'}
+
+### Level-Set Segmentation
+
+There are a variety of level-set based segmentation filter available in ITK:
```
Comment (drcandacemakedamoore) [inline episodes/simpleitk.md:1014]: What is missing here is a sentence to tell me what level set segmentation means. I still have only a vague idea that it has something to do with topology. If it's too complicated to explain, maybe a link to a paper or textbook.

**Review by drcandacemakedamoore — CHANGES_REQUESTED** [review]
There are some comments to consider I have placed . The main issue is that the code at the point indicated stops my kernel. Please go ahead and make a conda-environment file, so I can consider if this is related to my environment in some way. If you run your cells top to bottom, if your kernel intact?

**Comment by drcandacemakedamoore** [comment]
@gcroci2 I added the bell boxes.in a new commit ([12ceeb0](https://github.com/esciencecenter-digital-skills/medical-image-processing/pull/12/commits/12ceeb06f574e10351d813b661a8d13cfde60dbf)).  Feel free to change them a bit.

File: episodes/simpleitk.md [hunk 2]
```diff
@@ -0,0 +1,1198 @@
...
+nda_zslices = sitk.GetArrayViewFromImage(sitk.Tile(img_zslices, [1,0]))
+
+fig, (ax1, ax2, ax3) = plt.subplots(1,3)
+ax1.imshow(nda_xslices, cmap=plt.cm.Greys_r)
+ax2.imshow(nda_yslices, cmap=plt.cm.Greys_r)
+ax3.imshow(nda_zslices, cmap=plt.cm.Greys_r)
+ax1.set_title('X slices')
+ax2.set_title('Y slices')
+ax3.set_title('Z slices')
+```
```
[the remark this answers is not in the document]
Comment (gcroci2) [reply episodes/simpleitk.md:243]: I rerun the code and it doesn't break for me :)

**Comment by gcroci2** [comment]
Thanks for the suggestions, I added them :) @drcandacemakedamoore 

As I answered above, the code does not break for me. I don't have anything special in my env, only sitk and numpy installed, and I am using python 3.10.

**Review by drcandacemakedamoore — APPROVED** [review]

**Pull request merged**